components/canvas_traits/webgl.rs

```diff
@@ -175,10 +175,25 @@ impl WebGLMsgSender {
     pub fn send_dom_to_texture(&self, command: DOMToTextureCommand) -> WebGLSendResult {
         self.sender.send(WebGLMsg::DOMToTextureCommand(command))
     }
+
+    pub fn webxr_external_image_api(&self) -> impl webxr_api::WebGLExternalImageApi {
+        SerializableWebGLMsgSender {
+            ctx_id: self.ctx_id,
+            sender: self.sender.to_ipc(),
+        }
+    }
+}
+
+// WegGLMsgSender isn't actually serializable, despite what it claims.
+#[derive(Clone, Debug, Deserialize, MallocSizeOf, Serialize)]
+struct SerializableWebGLMsgSender {
+    ctx_id: WebGLContextId,
+    #[ignore_malloc_size_of = "channels are hard"]
+    sender: IpcSender<WebGLMsg>,
 }
 
 #[typetag::serde]
-impl webxr_api::WebGLExternalImageApi for WebGLMsgSender {
+impl webxr_api::WebGLExternalImageApi for SerializableWebGLMsgSender {
```

Comment:
You're still sending `sender`, a webgl_channel, over `self.sender`, an IPC channel here

Reply:
We'll need a separate lock API that works exclusively with IPC

Reply:
Wrote the fix and stuffed it in https://github.com/servo/servo/pull/23731